plugins/com.google.cloud.tools.eclipse.appengine.libraries.ui/src/com/google/cloud/tools/eclipse/appengine/libraries/ui/CloudLibrariesPage.java

```diff
@@ -103,13 +103,15 @@ public void initialize(IJavaProject project, IClasspathEntry[] currentEntries) {
       return null;
     }
 
+    // todo how do we get a real monitor here?
+    NullProgressMonitor monitor = new NullProgressMonitor();
```

Comment:
I don't think we can get the corresponding monitor.  But you can fake one as SubMonitor will handle a `null` monitor: `SubMonitor monitor = SubMonitor.convert(null /*monitor*/, <WORK>);`.

Reply:
Interesting. Does that produce a real monitor that shows feedback to the user and enables cancellation or is it a NullProgressMonitor alternative?